virtcontainers/sandbox.go

```diff
@@ -906,8 +906,19 @@ func (s *Sandbox) Delete() error {
 }
 
 func (s *Sandbox) createNetwork() error {
+	var netNsPath string
+	var netNsCreated bool
+	var err error
+
+	//rollback the NetNs when createNetwork failed
+	defer func() {
+		if err != nil && netNsPath != "" && netNsCreated {
```

Comment:
To make the check work in a defer closure, you need to make sure `err` is never redefined in this function.

Also the error from the last line `return s.storage.storeSandboxNetwork(s.id, networkNS)` should also be checked.

Reply:
Yeah, I had cleanup the err redefined lines and checked the last return code.

Thanks

Fupan